book/chapters/chp1/main.tex

```diff
@@ -143,7 +143,7 @@ \subsection*{Poisson}
 %\end{equation}
 %with $v$ a test function assumed to be zero on $\Gamma_D$. Solving the variational problem requires the assembly of a stiffness matrix on the left side, as well as a load vector on the right side. The dominant step in assembly is the stiffness matrix.
 
-Here we consider the problem of assembling a stiffness matrix for the solution of the Poisson equation on the unit cube. We use linear Lagrange finite elements on four different uniform, tetrahedral meshes ranging in size from about 1.3 to 16.5 million elements.  For each mesh, Table~\ref{tab:poisson_results} reports the throughput in millions of degrees of freedom per second (MDOF/s) for assembling the stiffness matrix with cuDOLFINx on a Hopper  GPU and with DOLFINx on a Grace CPU. Examining the results, the GPU doesn't appear fully saturated for the first two meshes, both of which have significantly fewer than one million degrees of freedom. This suggests that problems with approximately one million degrees of freedom or greater are needed to maximize the benefits of GPU acceleration. In the original exploration of GPU acceleration for FEniCSx, on which this work is based, a throughput of 189\,MDOF/s was reported for the Poisson assembly problem using the Uniform 3 mesh on an NVIDIA V100 GPU \citep{trotter2023targeting}. In this study, we attained a throughput of 376\,MDOF/s for the same mesh with the GH200 GPU, approximately twice the reported result for the V100. However, this is unsurprising given that the V100 has only 80 SMs compared to 132 for the newer Hopper GPU, as well as a lower clock speed (1.2 MHz to 1.9 MHz).
+Here we consider the problem of assembling a stiffness matrix for the solution of the Poisson equation on the unit cube. We use linear Lagrange finite elements on four different uniform, tetrahedral meshes ranging in size from about 1.3 to 16.5 million elements.  For each mesh, Table~\ref{tab:poisson_results} reports the throughput in millions of degrees of freedom per second (MDOF/s) for assembling the stiffness matrix with cuDOLFINx on a Hopper  GPU and with DOLFINx on a Grace CPU. Examining the results, the GPU doesn't appear fully saturated for the first two meshes, both of which have significantly fewer than one million degrees of freedom. This suggests that problems with approximately one million degrees of freedom or greater are needed to maximize the benefits of GPU acceleration. In the original exploration of GPU acceleration for FEniCSx, on which this work is based, a throughput of 189\,MDOF/s was reported for the Poisson assembly problem using the Uniform 3 mesh on an NVIDIA V100 GPU \citep{trotter2023targeting}. In this study, we achieved a throughput of 376 MDOF/s on the GH200 GPU, roughly twice that of the V100. This result is expected, as the GH200 offers higher memory bandwidth and FLOPS. Further investigation will be conducted to explore the underlying factors behind these performance gains.
```

Comment:
Feel free to add this back, though I’m not certain it’s the exact reason. I believe it’s worth investigating further. We could check whether it’s ALU-bound, which would suggest that more SMs and higher frequency could be contributing factors. Otherwise, if the L2 cache is the bottleneck, that might not be true. Also, the bottleneck could vary across different GPUs.